### PR TITLE
Replaced borderSide attribute with border attribute in BarChartRodData

### PR DESCRIPTION
**Pull request by MagdyYacoub1**
Resolve issue #1194

Replaced borderSide attribute with border attribute in [BarChartRodData](https://github.com/imaNNeo/fl_chart/blob/master/repo_files/documentations/bar_chart.md#barchartroddata)

This enables specifying each border side separately.

### CODE REVIEW

**Comment by imaNNeo** [comment]
I think you need to write some unit-tests

**Comment by imaNNeo** [comment]
It seems there are some conflicts, please rebase your branch again

**Comment by MagdyYacoub1** [comment]
I'm sorry for not getting back to you sooner, but I was away for the past two weeks.

I resolved the conflicts. please, try again now.

**Comment by imaNNeo** [comment]
Hi, sorry for my late response. There are new conflicts.
It would be great if you fix them again.
Thanks!

**Comment by MagdyYacoub1** [comment]
I resolved recent conflicts. Please provide me with some insights on why these conflicts happen so I can avoid that in the future. 
Thanks in advance. @imaNNeo

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 * **BUGFIX** (by @imaNNeo) Fix PieChart crash on web-renderer html by ignoring `sectionsSpace` when `Path.combine()` does not work (it's flutter engine [issue](https://github.com/flutter/flutter/issues/44572)), #955
 * **BUGFIX** (by @imaNNeo) Fix ScatterChart long-press interaction bug (disappears when long-pressing on the chart), #1318
 * **FEATURE** (by @imaNNeo) Upgrade dart version to [3.0](https://dart.dev/resources/dart-3-migration)
+* **BREAKING** (by @MagdyYacoub1) Replace the borderSide parameter for the BarChart with border parameter of type Border , #1194
```
Comment (imaNNeo) [inline CHANGELOG.md:9]: It is in the wrong place, you need to put it in the `## nextVersion` section

**Comment by imaNNeo** [comment]
There is an issue with BarSample 5
Please check it out, It will not happen in the `master` branch.
But it happens in your branch.
(There might be an issue with the canvas in your changes)

https://github.com/imaNNeo/fl_chart/assets/7009300/154cc07c-40e5-42c2-8851-dc9e3690a282

**Comment by imaNNeo** [comment]
> I resolved recent conflicts. Please provide me with some insights on why these conflicts happen so I can avoid that in the future. Thanks in advance. @imaNNeo

Because we are constantly update the `master` branch (by pushing new commits). So you need to update your branch with the master.
The best way to do that is rebasing. Please read our [contributing guidelines](https://github.com/imaNNeo/fl_chart/blob/master/CONTRIBUTING.md#keep-your-branch-updated)

**Pull request closed without merging**